src/opera/util/metfile.py

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright 2022, by the California Institute of Technology.
+# ALL RIGHTS RESERVED.
+# United States Government sponsorship acknowledged.
+# Any commercial use must be negotiated with the Office of Technology Transfer
+# at the California Institute of Technology.
+# This software may be subject to U.S. export control laws and regulations.
+# By accepting this document, the user agrees to comply with all applicable
+# U.S. export laws and regulations. User has the responsibility to obtain
+# export licenses, or other export authority as may be required, before
+# exporting such information to foreign countries or providing access to
+# foreign persons.
+#
+
+"""
+=========
+metfile.py
+=========
+
+Creates .met file for the OPERA PGEs.
+
+This module is adapted for OPERA from the NISAR PGE R2.0.0 util/metfile.py
+Original Author: Alice Stanboli
+Adapted By: Jim Hofman
+
+"""
+
+
+import json
+import jsonschema
+import os
+
+
+class MetFile:
+    """Class read and write .met files."""
+
+    def __init__(self, met_file_name, met_dict={}):
+        """
+        Constructor get filename and initialize dictionary
+
+        Parameters
+        ----------
+        met_file_name : str, required
+            Name of the file that will be written to or read.
+
+        """
+
+        self._met_file = met_file_name
+        self.met_dict = met_dict
+        self.combined_error_msg = None
+
+    def set_key_value(self, key_name, val):
```

Comment:
Since this method is basically an alias for `dict.__setitem__`, it might be neat to replace this method with an actual ``__setitem__`` so we could use `MetFile` objects like they were dictionaries themselves.

Reply:
Done